grammar.js

```diff
@@ -826,6 +841,16 @@ module.exports = grammar(C, {
       )
     ),
 
+    requires_clause: $ => seq('requires', $._expression),
+    requires_expression: $ => seq('requires', optional($.parameter_list), $.requirement_seq),
+    requirement_seq: $ => seq('{', repeat($._requirement), '}'),
+    _requirement: $ => choice($.simple_requirement, $.type_requirement, $.compound_requirement),
+
+    simple_requirement: $ => $.expression_statement,
+    type_requirement: $ => seq('typename', $.expression_statement),
+    compound_requirement: $ => seq("{", $._expression, "}", optional('noexcept'), optional($.return_type_requirement), ";"),
```

Comment:
Does `optional($.trailing_return_type)` not work here (in place of `optional($.return_type_requirement)`)? If we still want it to be named differently in the parse tree, we can create an alias.

`$._expression` matches a lot more so the parser could shrink I bit if that works.

Reply:
Tried this locally and it _slightly_ helps.

I also made `simple_requirement` an alias of `expression_statement` which saved some nodes in the parsed trees, so I may suggest that as well.

Reply:
:+1: thanks. That's a lot better that way.